### DISCUSSION TITLE
Why is there no startAfter and endBefore?

### DISCUSSION THREAD

**Discussion by ngotrongphuc** [invertase/react-native-firebase#5432] [Q&A] [answered]
I found that those two methods above don't exist in this library but they do exist in official documents of Firebase

**Comment by mikehardy** [comment, marked as answer]
Any time the upstream SDKs have a method and this module doesn't, it just means no one has implemented it yet. There are little API gaps all over like this, that are filled in by community members like yourself, or by Invertase as priorities and time allows. Right now we're pretty busy still handling the new features from the last native SDK releases (AppCheck module, Firestore Bundles etc) so unless you have time to implement it, these methods may not be covered for a little while.

If you do want to give it a shot though, the process basically looks like:

1) get the repo forked/cloned and do a test run of the tests app so you're ready to change things https://github.com/invertase/react-native-firebase/blob/master/tests/README.md
2) add the method signature to the packages index.d.ts and index.js file with a simple implementation that just throws an error
3) add a test for that method in the file that looks right in the e2e directory and if the test is marked as `it.only` vs `it` when you run the test app it will run just that test - it's really fast

Now you're set with a tight development loop - it goes fast from this point

4) pick your favorite platform (android or ios) and start with it by actually calling a new native method from javascript, and implement that native method also starting just as throwing an error - and make sure the e2e test gets the error
5) go to the upstream SDK documentation and pull their example code snippets / in combination with existing code you can put together the implementation and now the e2e test should pass
6) do the same for the other platform

make a PR with the result

**Reply by mikehardy** [reply]
worth noting as soon as you make a PR, we have a GitHub action that automatically makes a zip file with the patch-package formatted patches you need to integrate it into a work project, so you can integrate your work easily where you actually need it. That solves the problem of how to deal with getting the change compiled in a useful form since this is a monorepo and it's not easy otherwise
